src/index.ts

```diff
@@ -1 +1,326 @@
+
+    cmake: { path: (v) => `cmake/${v}/bin/cmake`, package: 'cmake' },
+    cpack: { path: (v) => `cmake/${v}/bin/cpack`, package: 'cmake' },
+    ninja: { path: (v) => `cmake/${v}/bin/ninja`, package: 'cmake' },
+    ctest: { path: (v) => `cmake/${v}/bin/ctest`, package: 'cmake' },
+
+    screenshot2: { path: (v) => `cmdline-tools/${v}/bin/screenshot2`, package: 'cmdline-tools' },
+    apkanalyzer: { path: (v) => `cmdline-tools/${v}/bin/apkanalyzer`, package: 'cmdline-tools' },
+    profgen: { path: (v) => `cmdline-tools/${v}/bin/profgen`, package: 'cmdline-tools' },
+    retrace: { path: (v) => `cmdline-tools/${v}/bin/retrace`, package: 'cmdline-tools' },
+    avdmanager: { path: (v) => `cmdline-tools/${v}/bin/avdmanager`, package: 'cmdline-tools' },
+    lint: { path: (v) => `cmdline-tools/${v}/bin/lint`, package: 'cmdline-tools' },
+    sdkmanager: { path: (v) => `cmdline-tools/${v}/bin/sdkmanager`, package: 'cmdline-tools' },
+} as const satisfies Record<string, { path: (version: string) => string; package: string }>;
+/* eslint-enable camelcase */
+
+/** The name of an Android development tool, either unversioned or versioned. */
+export type AndroidToolName = keyof typeof versionedAndroidTools | keyof typeof unversionedAndroidTools;
+/**
+ * An Android development tool, specified as either a name (which will cause the latest installed or available version
+ * to be used) or a name with an explicit version (only possible for versioned tools).
+ *
+ * The version in this context is _not_ the version of the tool itself, but rather the version of the package that
+ * contains the tool.
+ */
+export type AndroidTool = AndroidToolName | { tool: keyof typeof versionedAndroidTools; packageVersion: string };
+
+const platform = process.platform;
+if (platform !== 'win32' && platform !== 'darwin' && platform !== 'linux')
+    throw new Error(`Unsupported platform: "${platform}"`);
+
+const ensureSdkmanager = async () => {
+    const androidHome = await globalCacheDir('andromatic');
+
+    // Create license files just like Debian and Ubuntu do, see:
+    // https://github.com/tweaselORG/meta/issues/26#issuecomment-1535036229
+    const licensesDir = join(androidHome, 'licenses');
+    await fs.mkdir(licensesDir, { recursive: true });
+    const licenseHashes = {
+        'android-googletv-license': '601085b94cd77f0b54ff86406957099ebe79c4d6',
+        'android-sdk-arm-dbt-license': '859f317696f67ef3d7f30a50a5560e7834b43903',
+        'android-sdk-license': '24333f8a63b6825ea9c5514f83c2829b004d1fee',
+        'android-sdk-preview-license': '84831b9409646a918e30573bab4c9c91346d8abd',
+        'google-gdk-license': '33b6a2b64607f11b759f320ef9dff4ae5c47d97a',
+        'mips-android-sysimage-license': 'e9acab5b5fbb560a72cfaecce8946896ff6aab9d',
+    };
+    for (const [license, hash] of Object.entries(licenseHashes)) {
+        const licenseFile = join(licensesDir, license);
+        await fs.writeFile(licenseFile, hash);
+    }
```

Comment:
I guess you shouldn't need to do this if the `licenses` folder exists already, do you? Since you run this in every other function, this might save some computation time.

Reply:
I don't think just checking for the folder is enough:

* What if one of the files got deleted?
* More importantly: The hashes can change in the future. If we only check for the folder existence, we won't ever update them.

The second problem also can't be solved by checking for each file's existence. I guess we could read them and compare the contents, which feels like it should still be a little quicker but this seems like hair splitting to me.

Buuuut, I guess we really only need to do this if we're installing…